oop1-construct.js

```diff
@@ -20,58 +66,99 @@ Create a class that can categorize at least 6 of the creatures that you recall o
 
 class Shape {
     //your code here...
+    constructor(name, sides, base, height, area, length, width) {
+        this.name = name;
+        this.sides = sides;
+        this.base = base;
+        this.height = height;
+        this.area = area;
+        this.length = length;
+        this.width = width;
+    }
+
+
 }
+class Triangle extends Shape {
+    constructor(name, sides, base, height, area) {
+        super(name, sides, base, height, area);
+    }
+
 
+    HeronsCalc() {
+        let heronsStart = this.sides.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
+        let variableS = heronsStart / 2;
+        let triangleArea = Math.sqrt(variableS * ((variableS - this.sides[0]) * (variableS - this.sides[1]) * (variableS - this.sides[2])));
```

Comment:
When you have long formulas like this, you can use whitespace to make it more readable. Unfortunately the comment system here doesn't like any whitespace I add, so I can't show you an example here, but if you hit return after `Math.sqrt(` it's easy to read and see that everything under it and until you have a ) on the same level is going to be affected by it.